### ISSUE TITLE
Doubt related to the paper 

### ISSUE THREAD

**Issue by bhattg** [BenMussay/Data-Independent-Neural-Pruning-via-Coresets#2]
Dear Authors, 

My main question is related to the upper bound on the norm of query space (\beta, using the notations from the paper). When translating this to the neural network pruning, this would mean that while pruning the layer L, \beta will bound the norm of activations (which will be the queries) of the layer L-1. Since the pruning is performed from the first hidden layer, in sequential order, this would mean that \beta will be the bound on new activations from layer L-1, after pruning. 

My doubts are the following:
While implementing the algorithm, how do you estimate this (new) upper bound? Is it evaluated mathematically (by assuming a bound on the input space, and then appropriately percolating this deep in the layers), or is it evaluated empirically?
In a case where it is evaluated empirically, is complete training data used to evaluate \beta, or a subset of training data is chosen?  
While reading the discussion here, I made a note that you mentioned using batch norm and input normalization. Therefore, to evaluate \beta, are you using the batch-norm's scaling (and additive constant)? If yes, then I anticipate that one might need to consider the probability of an event where the norm of the previous layer's pruned activations exceeds \beta (evaluated using batch-norm). 

Thanks!
